src/OmnichannelChatSDK.ts

```diff
@@ -922,7 +922,40 @@ class OmnichannelChatSDK {
         } else {
             await sessionInitPromise(); // Await the session initialization
         }
-        await Promise.all([messagingClientPromise(),attachmentClientPromise()]);
+
+        try {
+            await Promise.all([messagingClientPromise(),attachmentClientPromise()]);
+        } catch (error) {
+            // If conversation joining fails after conversation was created, clean up the conversation
+            if (error instanceof ChatSDKError &&
+                error.message === ChatSDKErrorName.MessagingClientConversationJoinFailure &&
+                !this.chatSDKConfig.useCreateConversation?.disable) {
+                try {
+                    const sessionCloseOptionalParams: ISessionCloseOptionalParams = {};
+                    if (this.authenticatedUserToken) {
+                        sessionCloseOptionalParams.authenticatedUserToken = this.authenticatedUserToken;
+                    }
+                    await this.OCClient.sessionClose(this.requestId, sessionCloseOptionalParams);
```

Comment:
Scenarios to consider:
1. Should we close a conversation if `liveChatContext` was passed? The conversation was already created in an older session
2. Should we close a conversation upon reconnecting to an existing persistent chat conversation?
3. Should we close a conversation upon reconnecting to an existing chat reconnect conversation?

A concern I've is we unexpectedly close a conversation that was not supposed to be closed. We need to ensure that we only close a conversation that was "freshly created"